langkit/compile_context.py

```diff
@@ -2116,13 +2128,16 @@ def dependencies(typ):
         # by accident.
         struct_types = CompiledTypeRepo.struct_types
         array_types = CompiledTypeRepo.array_types
+        iterator_types = CompiledTypeRepo.iterator_types
         CompiledTypeRepo.struct_types = None
         CompiledTypeRepo.array_types = None
+        CompiledTypeRepo.iterator_types = None
 
         # Sort the struct and array types by dependency order
```

Comment:
Please update the comment. No need to extend the enumeration: just say for instance: “Sort all composite types by dependency order”.